src/Microsoft.Health.Dicom.Core/Features/ExtendedQueryTag/AddExtendedQueryTagService.cs

```diff
@@ -2,49 +2,44 @@
 {
     public class AddExtendedQueryTagService : IAddExtendedQueryTagService
     {
-        private readonly IExtendedQueryTagStore _extendedQueryTagStore;
-        private readonly IExtendedQueryTagEntryValidator _extendedQueryTagEntryValidator;
-        private readonly int _maxAllowedCount;
-
-        public AddExtendedQueryTagService(
-            IStoreFactory<IExtendedQueryTagStore> extendedQueryTagStoreFactory,
-            IExtendedQueryTagEntryValidator extendedQueryTagEntryValidator,
-            IOptions<ExtendedQueryTagConfiguration> extendedQueryTagConfiguration)
+        private readonly IExtendedQueryTagEntryValidator _tagValidator;
+        private readonly IDicomOperationsClient _client;
+
+        public AddExtendedQueryTagService(IExtendedQueryTagEntryValidator tagValidator, IDicomOperationsClient client)
         {
-            EnsureArg.IsNotNull(extendedQueryTagStoreFactory, nameof(extendedQueryTagStoreFactory));
-            EnsureArg.IsNotNull(extendedQueryTagEntryValidator, nameof(extendedQueryTagEntryValidator));
-            EnsureArg.IsNotNull(extendedQueryTagConfiguration?.Value, nameof(extendedQueryTagConfiguration));
+            EnsureArg.IsNotNull(tagValidator, nameof(tagValidator));
+            EnsureArg.IsNotNull(client, nameof(client));
 
-            _extendedQueryTagStore = extendedQueryTagStoreFactory.GetInstance();
-            _extendedQueryTagEntryValidator = extendedQueryTagEntryValidator;
-            _maxAllowedCount = extendedQueryTagConfiguration.Value.MaxAllowedCount;
+            _tagValidator = tagValidator;
+            _client = client;
         }
 
-        public async Task<AddExtendedQueryTagResponse> AddExtendedQueryTagAsync(IEnumerable<AddExtendedQueryTagEntry> extendedQueryTags, CancellationToken cancellationToken)
+        public async Task<AddExtendedQueryTagResponse> AddExtendedQueryTagsAsync(IEnumerable<AddExtendedQueryTagEntry> extendedQueryTags, CancellationToken cancellationToken)
         {
-            _extendedQueryTagEntryValidator.ValidateExtendedQueryTags(extendedQueryTags);
+            EnsureArg.IsNotNull(extendedQueryTags, nameof(extendedQueryTags));
 
-            IEnumerable<AddExtendedQueryTagEntry> result = extendedQueryTags.Select(item => item.Normalize());
+            _tagValidator.ValidateExtendedQueryTags(extendedQueryTags);
+            List<AddExtendedQueryTagEntry> normalized = extendedQueryTags
+                .Select(item => item.Normalize())
+                .ToList();
 
-            await _extendedQueryTagStore.AddExtendedQueryTagsAsync(result, _maxAllowedCount, cancellationToken);
+            EnsureArg.HasItems(normalized, nameof(extendedQueryTags));
 
-            // Current solution is synchronous, no job uri is generated, so always return blank response.
-            return new AddExtendedQueryTagResponse();
+            return new AddExtendedQueryTagResponse(await _client.StartExtendedQueryTagAdditionAsync(normalized, cancellationToken));
```

Comment:
Should we separate the responsibility of the Job to just update the status?
This creates the entry, marks it say "created", passes the contract information like say watermark and tags to the job. The job then just runs it and just updates the status when done.

Reply:
As discussed, we can do either way. But in a follow-up PR, I think I'll move the addition of the tag to DICOM server